llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1437,6 +1437,49 @@ void VPlanTransforms::truncateToMinimalBitwidths(
          "some entries in MinBWs haven't been processed");
 }
 
+/// Remove BranchOnCond recipes with constant conditions together with removing
+/// dead edges to their successors.
```

Comment:
The successors across the removed edges are assumed to have ResumePhi recipes, which are fixed.

Reply:
Yep

Reply:
I am not sure, they are expected to be valid coming in and the transform will keep them valid.